### PR TITLE
bugfix/issue-17 fvar specialization fix for trace_quad_form

### PR DESCRIPTION
**Pull request by rtrangucci**
#### Summary:

Introduces Matrix<double, -1, -1>, Matrix<fvar<T>, -1, -1> specialization of trace_quad_form

#### Intended Effect:

Admits higher-order autodiff of trace_quad_form for mixed type arguments (e.g. Matrix<double, -1, -1>, Matrix<fvar<var>, -1, -1>). Allows for fvar<T> instantiation of multi_normal_prec_log.

#### How to Verify:

./runTests.py test/unit/math/mix/mat/fun/trace_quad_form_test.cpp
./runTests.py test/unit/math/fwd/mat/fun/trace_quad_form_test.cpp

These should fail in the current math branch.

#### Side Effects:

None.

#### Documentation:

None.

#### Reviewer Suggestions:

Anyone

### CODE REVIEW

File: stan/math/fwd/mat/fun/trace_quad_form.hpp [hunk 1]
```diff
@@ -26,6 +26,34 @@ namespace stan {
       return stan::math::trace(multiply(stan::math::transpose(B),
                                         multiply(A, B)));
     }
+
+    template<int RA, int CA, int RB, int CB, typename T>
+    inline stan::math::fvar<T>
+    trace_quad_form(const Eigen::Matrix<stan::math::fvar<T>, RA, CA> &A,
+                    const Eigen::Matrix<double, RB, CB> &B) {
+      using stan::math::multiply;
```
Comment (syclik) [inline stan/math/fwd/mat/fun/trace_quad_form.hpp]: remove namespace qualification.

File: test/unit/math/mix/mat/fun/trace_quad_form_test.cpp [hunk 2]
```diff
@@ -110,6 +110,89 @@ TEST(AgradMixMatrixTraceQuadForm, mat_fv_1st_deriv) {
...
+          5,  2;
+  ad << 2.0,  3.0, 4.0,   5.0, 
+  6.0, 10.0, 2.0,   2.0,
+  7.0,  2.0, 7.0,   1.0,
+  8.0,  2.0, 1.0, 112.0;
+  
+
+  // fvar<var> - fvar<var>
+  res = trace_quad_form(ad,bd);
+}
```
Comment (syclik) [inline test/unit/math/mix/mat/fun/trace_quad_form_test.cpp]: add some sort of expect test for these new test cases.

**Comment by syclik** [comment]
@rtrangucci, I'm going to reindent the code, which will kick off tests once more. If it passes, we'll get it in.

**Comment by rtrangucci** [comment]
Sounds good!

Rob

**Comment by syclik** [comment]
Jenkins, retest this please.

**Pull request merged**